client/src/main/scala/com.gu.contentapi.client/model/PaginatedApiResponse.scala

```diff
@@ -0,0 +1,35 @@
+package com.gu.contentapi.client
+
+import com.gu.contentapi.client.model.v1._
+import com.gu.contentatom.thrift.Atom
+
+/** Typeclass witnessing how to extract the id for
+  * a paginated query
+  */
+trait PaginatedApiResponse[Response] {
+  /** the id for a next query, if any */
+  def getNextId: Response => Option[String]
+}
+
+/** Instances of [[PaginatedApiResponse]] for response types
+  * potentially containing pages of results
+  */
+object PaginatedApiResponse {
+
+  implicit val searchResponse = new PaginatedApiResponse[SearchResponse] {
+    def getNextId = r => if (r.results.length < r.pageSize) None else r.results.lastOption.map(_.id)
+  }
+
+  implicit val itemResponse = new PaginatedApiResponse[ItemResponse] {
+    def getNextId = r => if (r.results.exists(res => r.pageSize.exists(res.length < _))) None else r.results.flatMap(_.lastOption.map(_.id))
+  }
+
+  implicit val tagsResponse = new PaginatedApiResponse[TagsResponse] {
+    def getNextId = r => if (r.results.length < r.pageSize) None else r.results.lastOption.map(_.id)
+  }
+
+  implicit val atomsResponse = new PaginatedApiResponse[AtomsResponse] {
+    def getNextId = r => if (r.results.length < r.pageSize) None else r.results.lastOption.map(_.id)
```

Comment:
3 of the responses have this method duplicated. Can we extract it to a common method?

Reply:
Alas, no, I don't think so, the only super type of each is `Object` or something like that.